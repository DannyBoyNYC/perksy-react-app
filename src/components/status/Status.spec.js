const React = require('react');

const Status = require('./index').default;

describe('<Status.* />', () => {
  let component;
  const requiredProps = {
    code: 'draft'
  };

  describe('<Status.Button />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Status.Button {...requiredProps} />,
        'Status'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('renders prefix text before status if given', () => {
      component.setProps({
        prefix: 'Before words '
      });
      expect(component).toHaveText('Before words Draft');
    });

    it('renders suffix text after status if given', () => {
      component.setProps({
        suffix: ' after words'
      });
      expect(component).toHaveText('Draft after words');
    });
  });

  describe('<Status.Title />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(<Status.Title code="live" />, 'Status');
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
