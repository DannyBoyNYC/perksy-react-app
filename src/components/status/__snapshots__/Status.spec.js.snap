// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status.* /> <Status.Button /> should render correctly with all required props 1`] = `
<div
  className="Status-root-0-1-1 Status-button-0-1-5"
>
  <span
    className="Status-code-0-1-2"
  >
    Draft
  </span>
</div>
`;

exports[`<Status.* /> <Status.Title /> should render correctly with all required props 1`] = `
<div
  className="Status-root-0-1-1 Status-title-0-1-6"
>
  <span
    className="Status-code-0-1-2"
  >
    Active
  </span>
</div>
`;
